pkg/model/dynamic-plugins.go

```diff
@@ -203,6 +197,90 @@ func (p *DynamicPlugins) pluginsFromConfigMap() ([]DynaPlugin, error) {
 	return pluginsConfig.Plugins, nil
 }
 
+func (p *DynamicPlugins) mergeWith(specData string) (string, error) {
+
+	if p.ConfigMap == nil {
+		return "", fmt.Errorf("dynamic plugins ConfigMap is not set")
+	}
+	modelData := p.ConfigMap.Data[DynamicPluginsFile]
+	var modelPluginsConfig, specPluginsConfig DynaPluginsConfig
+	if modelData != "" {
+		if err := yaml.Unmarshal([]byte(modelData), &modelPluginsConfig); err != nil {
+			return "", fmt.Errorf("failed to unmarshal model ConfigMap data: %w", err)
+		}
+	}
+	if specData != "" {
+		if err := yaml.Unmarshal([]byte(specData), &specPluginsConfig); err != nil {
+			return "", fmt.Errorf("failed to unmarshal spec ConfigMap data: %w", err)
+		}
+	}
+
+	// Merge Plugins by package field
+	pluginMap := make(map[string]DynaPlugin)
+	for _, plugin := range modelPluginsConfig.Plugins {
+		pluginMap[plugin.Package] = plugin
+	}
+	for _, plugin := range specPluginsConfig.Plugins {
+
+		if existingPlugin, found := pluginMap[plugin.Package]; found {
+			if plugin.PluginConfig != nil {
+				existingPlugin.PluginConfig = plugin.PluginConfig
+			}
+			if len(plugin.Dependencies) > 0 {
+				existingPlugin.Dependencies = plugin.Dependencies
+			}
+			if plugin.Integrity != "" {
+				existingPlugin.Integrity = plugin.Integrity
+			}
+			existingPlugin.Disabled = plugin.Disabled
+			pluginMap[plugin.Package] = existingPlugin
+		} else {
+			// If the plugin is not found in model, add it from spec
+			if !plugin.Disabled {
+				pluginMap[plugin.Package] = plugin
+			}
+		}
+	}
+	mergedPluginsConfig := modelPluginsConfig
+	mergedPluginsConfig.Plugins = make([]DynaPlugin, 0, len(pluginMap))
```

Comment:
I believe this may create a shallow copy, which could lead to unintended mutations of the original config?
```suggestion
    mergedPluginsConfig := DynaPluginsConfig{
        Includes: make([]string, 0, len(includeSet)),
        Plugins:  make([]DynaPlugin, 0, len(pluginMap)),
    }
```

Reply:
so currently both `mergedPluginsConfig` and `modelPluginsConfig`  share the same [`includes`](https://github.com/gazarenkov/rhdh-operator/blob/external-config/pkg/model/dynamic-plugins.go#L49) slice in memory. so during include merging this could modify the original [`modelPluginsConfig.Includes`](https://github.com/gazarenkov/rhdh-operator/blob/external-config/pkg/model/dynamic-plugins.go#L49) if multiple reconciliations happen simultaneously. But I think we are actually fine as we create a new slice immediately.